src/android/SonectCordovaPlugin.java

```diff
@@ -1,27 +1,68 @@
 package ch.sonect.sdk;
 
+import android.util.Log;
+
 import org.apache.cordova.CallbackContext;
 import org.apache.cordova.CordovaArgs;
 import org.apache.cordova.CordovaPlugin;
+import org.json.JSONArray;
 import org.json.JSONException;
+import org.json.JSONObject;
 
 import java.util.ArrayList;
 
 public class SonectCordovaPlugin extends CordovaPlugin {
     @Override
     public boolean execute(String action, CordovaArgs args, CallbackContext callbackContext) throws JSONException {
         if ("present".equalsIgnoreCase(action)) {
-            ArrayList<SDKEntryPointActivity.PaymentMethodReference> pms = new ArrayList<>(1);
-            // TODO define JS interface with NEON and then provide valid payment method
+            processPresentCommand(args);
+        }
+        return true;
+    }
+
+    private void processPresentCommand(CordovaArgs args) throws JSONException {
+        ArrayList<SDKEntryPointActivity.PaymentMethodReference> pms = new ArrayList<>();
+
+        JSONObject credentials = args.getJSONObject(0);
+        JSONArray paymentMethods = args.getJSONArray(1);
+        JSONObject theme = args.getJSONObject(2);
+
+        boolean isWithdrawInProgress = !args.isNull(3);
+        JSONObject paymentRef = null;
+        if (isWithdrawInProgress) {
+            paymentRef = args.getJSONObject(3);
+        }
+
+        for (int i = 0; i < paymentMethods.length(); i++) {
+            JSONObject pm = paymentMethods.getJSONObject(i);
             pms.add(new SDKEntryPointActivity.PaymentMethodReference(
-                            "My payment method", R.drawable.ic_visa, "**** 1234", 302.44f
+                            pm.getString("name"), R.drawable.ic_sonect_web_bank, pm.getString("detailDescription"), (float) pm.getDouble("funds")
                     )
             );
+        }
+
+        if (isWithdrawInProgress) {
+            String uniqueID = null;
+            if (paymentRef.has("uniqueIdentifier")) {
+                uniqueID = paymentRef.getString("uniqueIdentifier");
+            }
+            SDKEntryPointActivity.Companion.startWithdraw(cordova.getActivity(),
+                    new SonectSDK.Config.UserCredentials(credentials.getString("userId"),
+                            credentials.getString("token"),
+                            credentials.getString("signature"), "eu", null),
+                    new SDKEntryPointActivity.NewTransactionReference(paymentRef.getString("paymentReference"), uniqueID, "30", "CHF"),
```

Comment:
AFAIK it's not enough to get payment ref only to create soncet transaction. We also need value+currency.

Reply:
Value / Currency is "remembered" in the payment method that is processing this transaction, to confirm that transaction that is in process we just need a payment reference from the bank.

Reply:
In case of android we close and SDK and loose the context. Maybe we could try to cache the values in preferences. So there are 2 options
1) cache values in preferences
2) force outer user to provide this info

In case of caching need to think about edge cases when cache could be invalidated. I mean that if we detach sonect context from neon context maintanance could be simpler.